src/smartcontracts/nativeSerializer.ts

```diff
@@ -296,7 +296,8 @@ export namespace NativeSerializer {
             return new AddressValue(convertNativeToAddress(native, errorContext));
         }
         if (type instanceof BooleanType) {
-            return new BooleanValue(native);
+            const boolValue = native.toString().toLowerCase() === "true" || native.toString() === "1";
```

Comment:
:rocket:

Reply:
Wouldn't this be more appropriate in the "isTrue" evaluation?